### ISSUE TITLE
BUG: akash-hostname-operator stops processing requests on an error

### ISSUE THREAD

**Issue by arno01** [ovrclk/akash#1510]
`akash-hostname-operator` stops processing requests on an error.
In this particular case, the service name starting with a digit is causing this.
Service name should follow DNS-1035 standard.

This poses an issue - whoever deploys a bad SDL will cause provider unable to serve URI's for further deployments.

## Reproducer

1. deploy SDL manifest with the service name starting with a digit;
2. now deploy any good SDL manifest;

## Expecting

Both deployments should get working URI's.

## Actually

URI's are returning `404 Not Found`, despite the pods got deployed and are running.

`akash-hostname-operator` stops processing further requests on an error caused by a bad service name in the manifest of the first deployment.

`spec.rules[0].http.paths[0].backend.service.name: Invalid value: \"2acoin\": a DNS-1035 label must consist ...`

Simplified:

```
$ kubectl -n akash-services logs $(kubectl -n akash-services get pods -l app=akash-hostname-operator -o jsonpath='{.items[0].metadata.name}') --tail=20
...
failed applying event
observation stopped
delaying
```

## Bad SDL

```
version: "2.0"

services:
  2acoin:
    image: ourlink/2acoin_xmrig:latest
    expose:
      - port: 8080
        as: 80
        to:
          - global: true
    env:
      - "ADDRESS=REDACTED.100000"
      - "POOL=us-central.2acoin.org:5555"
      - "WORKER=test02"
      - "ALGO=argon2/chukwa"
      - "TLS=false" #If supported by pool
      - "TLS_FINGERPRINT=" #Can be blank
      
profiles:
  compute:
    2acoin:
      resources:
        cpu:
          units: 0.25  # was 1
        memory:
          size: 2Gi
        storage:
          size: 1Gi
  placement:
    dcloud:
      attributes:
        host: akash
      signedBy:
        anyOf:
          - "akash1365yvmc4s7awdyj3n2sav7xfx76adc6dnmlx63"
      pricing:
        2acoin:
          denom: uakt
          amount: 100
          
deployment:
  2acoin:
    dcloud:
      profile: 2acoin
      count: 4
```

## akash-hostname-operator logs

```
kubectl -n akash-services logs $(kubectl -n akash-services get pods -l app=akash-hostname-operator -o jsonpath='{.items[0].metadata.name}') --tail=20
E[2022-01-30|15:45:14.758] observation stopped                          err="Ingress.extensions \"kjichb4p21bc7370snufdjh37o.ingr03.validatornode.com\" is invalid: spec.rules[0].http.paths[0].backend.service.name: Invalid value: \"2acoin\": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')"
I[2022-01-30|15:45:14.758] delaying
I[2022-01-30|15:45:17.758] starting observation
D[2022-01-30|15:45:17.765] identified existing hostname connection      hostname=qgt0sfuh6hdpbbser9if7h20vc.ingr03.validatornode.com lease=akash1h24fljt7p0nh82cq0za0uhsct3sfwsfu9w3c9h/4501139/1/1/akash1yqgj0fce4pyax2yhtl2cwl8arv9tlgt77dw3xv service=sifchain-ui port=80
D[2022-01-30|15:45:17.765] identified existing hostname connection      hostname=f5uc943h2he65dipug2ectmi9g.ingr03.validatornode.com lease=akash19xgz72f5772y5gf4xrfyjmu6wxhw5st703ccv8/4348965/1/1/akash1yqgj0fce4pyax2yhtl2cwl8arv9tlgt77dw3xv service=web port=80
D[2022-01-30|15:45:17.765] identified existing hostname connection      hostname=8riau11a2l9559n7e6bptlqj9o.ingr03.validatornode.com lease=akash19xgz72f5772y5gf4xrfyjmu6wxhw5st703ccv8/4501027/1/1/akash1yqgj0fce4pyax2yhtl2cwl8arv9tlgt77dw3xv service=web port=80
D[2022-01-30|15:45:17.765] identified existing hostname connection      hostname=www.mysite.com lease=akash19xgz72f5772y5gf4xrfyjmu6wxhw5st703ccv8/4501027/1/1/akash1yqgj0fce4pyax2yhtl2cwl8arv9tlgt77dw3xv service=web port=80
I[2022-01-30|15:45:17.769] starting hostname watch                      module=provider-cluster-kube resourceVersion=2071483
D[2022-01-30|15:45:17.770] apply event                                  event-type=add hostname=8riau11a2l9559n7e6bptlqj9o.ingr03.validatornode.com
D[2022-01-30|15:45:17.772] connecting                                   hostname=8riau11a2l9559n7e6bptlqj9o.ingr03.validatornode.com lease=akash19xgz72f5772y5gf4xrfyjmu6wxhw5st703ccv8/4501027/1/1/akash1yqgj0fce4pyax2yhtl2cwl8arv9tlgt77dw3xv service=web externalPort=80
D[2022-01-30|15:45:17.772] apply event                                  event-type=add hostname=f5uc943h2he65dipug2ectmi9g.ingr03.validatornode.com
D[2022-01-30|15:45:17.774] connecting                                   hostname=f5uc943h2he65dipug2ectmi9g.ingr03.validatornode.com lease=akash19xgz72f5772y5gf4xrfyjmu6wxhw5st703ccv8/4348965/1/1/akash1yqgj0fce4pyax2yhtl2cwl8arv9tlgt77dw3xv service=web externalPort=80
D[2022-01-30|15:45:17.774] apply event                                  event-type=add hostname=kjichb4p21bc7370snufdjh37o.ingr03.validatornode.com
D[2022-01-30|15:45:17.776] connecting                                   hostname=kjichb4p21bc7370snufdjh37o.ingr03.validatornode.com lease=akash19xgz72f5772y5gf4xrfyjmu6wxhw5st703ccv8/4501341/1/1/akash1yqgj0fce4pyax2yhtl2cwl8arv9tlgt77dw3xv service=2acoin externalPort=80
D[2022-01-30|15:45:17.776] hostname target is new, applying
D[2022-01-30|15:45:17.776] Updating ingress
E[2022-01-30|15:45:17.787] failed applying event                        err="Ingress.extensions \"kjichb4p21bc7370snufdjh37o.ingr03.validatornode.com\" is invalid: spec.rules[0].http.paths[0].backend.service.name: Invalid value: \"2acoin\": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')"
D[2022-01-30|15:45:17.787] hostname operator done
E[2022-01-30|15:45:17.787] observation stopped                          err="Ingress.extensions \"kjichb4p21bc7370snufdjh37o.ingr03.validatornode.com\" is invalid: spec.rules[0].http.paths[0].backend.service.name: Invalid value: \"2acoin\": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')"
I[2022-01-30|15:45:17.787] delaying
```

## checking nginx ingress controller

As akash hostname opreator stopped observing due to an error in bad SDL, it did not update the nginx.conf for a providerhost `n1f8s6p92h81f30su4l8990f04` of the next deployment:

```
$ kubectl -n ingress-nginx exec $(kubectl -n ingress-nginx get pods -l app.kubernetes.io/component=controller --output jsonpath='{.items[0].metadata.name}') -- cat nginx.conf | grep -i n1f8s6p92h81f30su4l8990f04

NO output as akash hostname operator borked because of a bad SDL mentioned before.
```

**Comment by arno01** [comment]
As a solution: a service name should be rejected by both, CLI & Server if it does not follow DNS-1035 standard:

`DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?'`

**Comment by hydrogen18** [comment]
I tried to deploy this 

```
version: "2.0"

services:
  2web:
    image: quay.io/ovrclk/demo-app
    expose:
      - port: 80
        as: 80
        http_options:
          max_body_size: 2097152
          next_cases:
            - off
        accept:
          - hello.localhost
        to:
          - global: true

profiles:
  compute:
    web:
      resources:
        cpu:
          units: 0.1
        memory:
          size: 16Mi
        storage:
          size: 128Mi
  placement:
    westcoast:
      attributes:
        region: us-west
      pricing:
        web: 
          denom: uakt
          amount: 1000

deployment:
  2web:
    westcoast:
      profile: web
      count: 1

```

I got this error in my hostname operator

```
D[2022-02-01|15:50:08.409] hostname target is new, applying             
D[2022-02-01|15:50:08.409] Updating ingress                             
E[2022-02-01|15:50:08.411] failed applying event                        err="Ingress.extensions \"hello.localhost\" is invalid: spec.rules[0].http.paths[0].backend.service.name: Invalid value: \"2web\": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')"
D[2022-02-01|15:50:08.411] hostname operator done                       
E[2022-02-01|15:50:08.411] observation stopped                          err="Ingress.extensions \"hello.localhost\" is invalid: spec.rules[0].http.paths[0].backend.service.name: Invalid value: \"2web\": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')"
I[2022-02-01|15:50:08.411] delaying                  
``` 

so it looks like this bug is legit

I see this in the CRD associated with it

```
 kubectl describe providerhosts.akash.network -n lease hello.localhost
Name:         hello.localhost
Namespace:    lease
Labels:       akash.network=true
              akash.network/lease.id.dseq=1
              akash.network/lease.id.gseq=1
              akash.network/lease.id.oseq=1
              akash.network/lease.id.owner=akash1gpupnfwhah2fqkmv33jw758m43xqplxay40rua
              akash.network/lease.id.provider=akash12glmhunqz238eakltjpnetg4cy53udxsmerqk0
Annotations:  <none>
API Version:  akash.network/v1
Kind:         ProviderHost
Metadata:
  Creation Timestamp:  2022-02-01T21:49:41Z
  Generation:          1
  Managed Fields:
    API Version:  akash.network/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:labels:
          .:
          f:akash.network:
          f:akash.network/lease.id.dseq:
          f:akash.network/lease.id.gseq:
          f:akash.network/lease.id.oseq:
          f:akash.network/lease.id.owner:
          f:akash.network/lease.id.provider:
      f:spec:
        .:
        f:dseq:
        f:external_port:
        f:gseq:
        f:hostname:
        f:oseq:
        f:owner:
        f:provider:
        f:service_name:
    Manager:         akash
    Operation:       Update
    Time:            2022-02-01T21:49:41Z
  Resource Version:  1193
  UID:               e7b927a4-1f34-4053-81ce-eda07e36e12a
Spec:
  Dseq:           1
  external_port:  80
  Gseq:           1
  Hostname:       hello.localhost
  Oseq:           1
  Owner:          akash1gpupnfwhah2fqkmv33jw758m43xqplxay40rua
  Provider:       akash12glmhunqz238eakltjpnetg4cy53udxsmerqk0
  service_name:   2web
```

Also the provider can't even create the service because the name is invalid

```
E[2022-02-01|15:49:41.285] applying local service                       module=provider-cluster-kube err="Service \"2web\" is invalid: metadata.name: Invalid value: \"2web\": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')" lease=akash1gpupnfwhah2fqkmv33jw758m43xqplxay40rua/1/1/1/akash12glmhunqz238eakltjpnetg4cy53udxsmerqk0 service=2web
```

**Comment by hydrogen18** [comment]
Should be fixed in latest 0.14.x RC

**Comment by hydrogen18** [comment]
@arno01 had anymore problems along these lines?
